### PR TITLE
487 add a ci-test to check raising error on eth_estimateGas when emulate gives revert

### LINKED ISSUE #487: Gas estimate with contract which do "revert" throw exception
In web3.py contract deployment uses two methods:
1. eth_estimateGas with contract instruction
2. deploy instruction.

If contract "reverted", proxy throw exception in log and revert error in response for estimateGas, but right behavior - return gas estimate.

```python
2022-01-25 11:14:57,472 - pid:22 [D] handle_request:645 - >>> 140379083720448 0x7fac8e832640 {"jsonrpc": "2.0", "id": 0, "result": {"gasUsed": "0x0", "hash": "0x214db04188f372129759203555a04631740b751fe16bf9641c53724c55574a8e", "number": "0xbe020", "parentHash": "0x2417754542879e51b7c7c70691f7d672573ba195f5b9e8faeb7b52ec3f047621", "timestamp": "0x61efdbb0", "transactions": [], "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "gasLimit": "0x6691b7"}} eth_getBlockByNumber resp_time_ms= 11.372089385986328
2022-01-25 11:14:57,473 - pid:22 [I] access_log:226 - 172.21.0.1:35872 - POST /solana - 13.21 ms
2022-01-25 11:14:57,575 - pid:30 [D] handle_request:624 - <<< 140379092559616 0x7fac8e7b2df0 {"jsonrpc": "2.0", "method": "eth_estimateGas", "params": [{"data": "0x73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220b0b6a0a6b54262aedb2d0b6e51ccf0bca6c08b196060a19204eb2b38d8074d3364736f6c63430006060033"}], "id": 1}
2022-01-25 11:14:57,580 - pid:30 [D] call:47 - Calling: neon-cli --commitment=recent --url http://release-solana:8899 --evm_loader=53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io -vvv emulate --token_mint HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU 0x0000000000000000000000000000000000000000 deploy 0x73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220b0b6a0a6b54262aedb2d0b6e51ccf0bca6c08b196060a19204eb2b38d8074d3364736f6c63430006060033
DEBUG - command_emulate(config=evm_loader=53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io, signer=Signer: BMp6gEnveANdvSvspESJUrNczuHz1GF5UQKjVLCkAZih, contract_id=None, caller_id=0x0000000000000000000000000000000000000000, data="73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220b0b6a0a6b54262aedb2d0b6e51ccf0bca6c08b196060a19204eb2b38d8074d3364736f6c63430006060033", value=Some(0))
DEBUG - get_ether_account_nonce data = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 37, 45, 67, 210, 189, 245, 240, 224, 140, 252, 41, 52, 109, 79, 63, 121, 63, 95, 27, 2, 84, 201, 181, 46, 148, 147, 161, 76, 41, 210, 247, 45, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
DEBUG - Caller: ether 0x0000…0000, solana 8ufd9goJkjJ776CWUXufi7ZpN4JiAJ2pEkk5DnYg8RS4
DEBUG - Caller trx_count: 0
DEBUG - caller_token = 3W87onJDCZfYpRniw8JbXu4jKAKB8uyBe3NSE3iRVJrp
DEBUG - program_id to deploy: 0xbd770416a3345f91e4b34576cb804a576fa48eb1
backend::new
Got slot
Slot 778273
Got timestamp
timestamp 1643109296
DEBUG - Executor initialized
DEBUG - create_begin(storage.origin()=0x0000000000000000000000000000000000000000, data="73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220b0b6a0a6b54262aedb2d0b6e51ccf0bca6c08b196060a19204eb2b38d8074d3364736f6c63430006060033", value=Some(0))
create_begin gas_limit=50000000
create target_gas=Some(50000000)
Not found account for 0x0000000000000000000000000000000000000000 => 8ufd9goJkjJ776CWUXufi7ZpN4JiAJ2pEkk5DnYg8RS4
Account found
Account data len 256
Account owner 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io
code_account == Pubkey::new_from_array([0u8; 32])
SolanaAccount::new
  SolidityAccount::new solana_address=8ufd9goJkjJ776CWUXufi7ZpN4JiAJ2pEkk5DnYg8RS4
  SolidityAccount::new solana_address=8ufd9goJkjJ776CWUXufi7ZpN4JiAJ2pEkk5DnYg8RS4
Not found account for 0xbd770416a3345f91e4b34576cb804a576fa48eb1 => F9wD5bUpb8wHtgW37nTR3y2cqhahtPhCNA7KAK8iPQak
Account not found 0xbd77…8eb1
Account not found 0xbd77…8eb1
Not found account for 0xbd770416a3345f91e4b34576cb804a576fa48eb1 => F9wD5bUpb8wHtgW37nTR3y2cqhahtPhCNA7KAK8iPQak
Account not found 0xbd77…8eb1
Not found account for 0xbd770416a3345f91e4b34576cb804a576fa48eb1 => F9wD5bUpb8wHtgW37nTR3y2cqhahtPhCNA7KAK8iPQak
Account not found 0xbd77…8eb1
executor transfer from=0x0000…0000 to=0xbd77…8eb1 value=0
DEBUG - Execute done, exit_reason=Revert(Reverted), result=[]
DEBUG - 7 steps executed
DEBUG - used_gas=54133 refunded_gas=0
DEBUG - Call done
INFO - revert
INFO -
DEBUG - Not succeed execution
2022-01-25 11:14:57,644 - pid:30 [D] call_emulated:14 - Call emulated. contract_id: deploy, caller_id: 0x0000000000000000000000000000000000000000, data: 0x73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220b0b6a0a6b54262aedb2d0b6e51ccf0bca6c08b196060a19204eb2b38d8074d3364736f6c63430006060033, value: , return: {"accounts":[{"account":"8ufd9goJkjJ776CWUXufi7ZpN4JiAJ2pEkk5DnYg8RS4","address":"0x0000000000000000000000000000000000000000","code_size":null,"code_size_current":null,"contract":null,"new":false,"writable":false},{"account":"F9wD5bUpb8wHtgW37nTR3y2cqhahtPhCNA7KAK8iPQak","address":"0xbd770416a3345f91e4b34576cb804a576fa48eb1","code_size":null,"code_size_current":null,"contract":null,"new":true,"writable":false}],"exit_status":"revert","result":"","solana_accounts":[],"steps_executed":7,"token_accounts":[],"used_gas":54133}
 
2022-01-25 11:14:57,645 - pid:30 [D] check_emulated_exit_status:24 - Got revert call emulated result with data:
2022-01-25 11:14:57,645 - pid:30 [D] eth_estimateGas:124 - Exception on eth_estimateGas:
Traceback (most recent call last):
  File "/opt/proxy/plugin/solana_rest_api.py", line 599, in process_request
    response['result'] = method(*params)
  File "/opt/proxy/plugin/solana_rest_api.py", line 122, in eth_estimateGas
    return estimate_gas(self.client, self.signer, contract_id, EthereumAddress(caller_id), data, value)
  File "/opt/proxy/plugin/solana_rest_api_tools.py", line 125, in estimate_gas
    result = call_emulated(contract_id, str(caller_eth_account), data, value)
  File "/opt/proxy/common_neon/emulator_interactor.py", line 16, in call_emulated
    check_emulated_exit_status(result)
  File "/opt/proxy/common_neon/emulator_interactor.py", line 27, in check_emulated_exit_status
    raise EthereumError(code=3, message='')
proxy.common_neon.errors.EthereumError
2022-01-25 11:14:57,646 - pid:30 [D] handle_request:645 - >>> 140379092559616 0x7fac8e7b2df0 {"jsonrpc": "2.0", "id": 1, "error": {"code": 3, "message": ""}} eth_estimateGas resp_time_ms= 71.12741470336914
2022-01-25 11:14:57,647 - pid:30 [I] access_log:226 - 172.21.0.1:35876 - POST /solana - 72.78 ms
```

### PR DESCRIPTION
**Pull request by vasiliy-zaznobin**

### CODE REVIEW

File: proxy/testing/test_eth_estimateGas.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    # @unittest.skip("a.i.")
+    def test_01_check_do_revert(self):
+        print("\ntest_check_get_block_by_number")
+        try:
+            nonce = proxy.eth.get_transaction_count(proxy.eth.default_account)
+            trx_revert = self.reverting_contract.functions.do_revert().buildTransaction({'nonce': nonce})
+            print('trx_revert:', trx_revert)
+            trx_estimate_gas_response = proxy.eth.estimate_gas(trx_revert)
+            print('trx_estimate_gas_response:', trx_estimate_gas_response)
```
Comment (gigimon) [inline proxy/testing/test_eth_estimateGas.py:79]: Why in this test nothing checks by assert? This test only writes logs and nothing verify.
We need to check trx_estimate_gas_response and revert message

File: proxy/testing/test_eth_estimateGas.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            self.assertTrue(False)
+        except Exception as e:
+            print('type(e):', type(e))
+            print('e:', e)
+            self.assertTrue(True)
```
Comment (vasiliy-zaznobin) [inline proxy/testing/test_eth_estimateGas.py:84]: @gigimon this assert checks that an exception is raised

File: proxy/testing/test_eth_estimateGas.py (hunk 1 above)
Comment (vasiliy-zaznobin) [reply proxy/testing/test_eth_estimateGas.py:79]: See https://github.com/neonlabsorg/proxy-model.py/pull/500/files#r793458484

File: proxy/testing/test_eth_estimateGas.py (hunk 2 above)
Comment (gigimon) [reply proxy/testing/test_eth_estimateGas.py:84]: ok

**Review by gigimon — APPROVED** [review]

**Pull request merged**